File: emp-api/src/main/java/com/example/entities/Leave.java
package com.example.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(name = "leave_table")
public class Leave {
    @Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
private Long id;

    private String leaveType;
    private LocalDate startDate;
    private LocalDate endDate;
    private String status;


    @ManyToOne
    @JoinColumn(name = "registration_id", nullable = false)

    private Registration registration;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getLeaveType() {
        return leaveType;
    }

    public void setLeaveType(String leaveType) {
        this.leaveType = leaveType;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Registration getRegistration() {
        return registration;
    }

    public void setRegistration(Registration registration) {
        this.registration = registration;
    }
}

File: emp_ui/lib/models/leave.dart
class LeaveModel {
  final int? id;
  final String leaveType;
  final String startDate;
  final String endDate;
  final String status;
  final int registrationId;

  LeaveModel({
    this.id,
    required this.leaveType,
    required this.startDate,
    required this.endDate,
    required this.status,
    required this.registrationId,
  });

  factory LeaveModel.fromJson(Map<String, dynamic> json) {
    return LeaveModel(
      id: json['id'],
      leaveType: json['leaveType'],
      startDate: json['startDate'],
      endDate: json['endDate'],
      status: json['status'],
      registrationId: json['registration']['id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "leaveType": leaveType,
      "startDate": startDate,
      "endDate": endDate,
      "status": status,
      "registration": {
        "id": registrationId
      }
    };
  }
}
